router/packetconn.go

```diff
@@ -46,14 +46,21 @@ func (r *Router) newLocalPeer() *peer {
 // was delivered using tree routing).
 func (r *Router) ReadFrom(p []byte) (n int, addr net.Addr, err error) {
 	var frame *types.Frame
+	readDeadline := time.Time{}
+	phony.Block(r, func() {
+		readDeadline = r._readDeadline
```

Comment:
I'm wondering whether this might be a good use for an `atomic` value, since otherwise I suspect we're creating contention around the router actor where none existed before.

Reply:
That's a good idea.